### PR TITLE
roll back dependency to fix warnings & vulnerabilities

### PR DESCRIPTION
**Pull request by danielortiz1982**
**Explain the details for making this change. What existing problem does the pull request solve?**
Rolling back dependency that contain vulnerabilities and warnings in the main branch. 

**Related github/jira issue (required)**:
No Issue

**Steps necessary to review your pull request (required)**:

Include:
- git checkout no-issue-fix-dependency-vulnerabilities-2022-6-9 -f
- rm -rf node_modules
- rm -rf package-lock.json
- npm i

Before:
Screen Shot 2022-06-09 at 8 36 45 AM

After:
Screen Shot 2022-06-09 at 8 51 42 AM

**Included in this Pull Request**:
- [ ] An e2e or functional test for the bug or feature.
- [ ] A note to the change log.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,24 +12,24 @@
   "scripts": {
     "build:dist": "npx rimraf build/dist/development && npx tsc && npx webpack --config ./webpack.config.prod.js",
     "build:dist:prod": "npx rimraf build/dist/production && npx tsc && npx webpack --config ./webpack.config.prod.js --mode production",
-    "build": "npx rimraf build/dist/development && node --max-old-space-size=8192 node_modules/webpack/bin/webpack --config ./webpack.config.dev.js",
-    "build:coverage": "npx rimraf build/dist/development && node --max-old-space-size=8192 node_modules/webpack/bin/webpack --config ./webpack.config.dev.js --env coverage",
+    "build": "npx rimraf build/dist/development && node --max-old-space-size=16384 node_modules/webpack/bin/webpack --config ./webpack.config.dev.js",
```
Comment (tmcconechy) [inline package.json:15]: This number is too big - it wont work unless the memory is this big. Should stay at 8192

File: package.json [hunk 2]
```diff
@@ -12,24 +12,24 @@
...
     "lint": "npx eslint src test && npx stylelint src/**/**/*.scss --config .stylelintrc.json && npm run mdlint && npx htmlhint 'src/**/*.html'",
     "mdlint": "npx remark . -f",
     "publish:dry-run": "npm run build:dist:prod && cp -f package-dist.json build/dist/production/package.json && cp -f README.md build/dist/production && cp -f LICENSE build/dist/production && cd build/dist/production && npm publish --dry-run",
     "publish:npm": "npm run build:dist:prod && cp -f package-dist.json build/dist/production/package.json && cp -f README.md build/dist/production && cp -f LICENSE build/dist/production && cd build/dist/production && npm publish",
-    "publish:debug": "npm run build:dist && cp -f package-dist.json build/dist/development/package.json && cp -f README.md build/dist/development && cp -f LICENSE build/dist/development && cd build/dist/development && npm publish",
+    "publish:debug": "npm run build:dist && cp -f package-dist.json build/dist/production/package.json && cp -f README.md build/dist/production && cp -f LICENSE build/dist/production && cd build/dist/production && npm publish",
```
Comment (tmcconechy) [inline package.json:26]: This should also stay for the prod builds to work

File: package.json [hunk 3]
```diff
@@ -12,24 +12,24 @@
...
     "test": "PERCY_LOGLEVEL=silent npx jest --coverage --runInBand --detectOpenHandles --forceExit",
     "test:watch": "PERCY_LOGLEVEL=silent npx jest --watchAll --coverage --detectOpenHandles --forceExit -- func",
     "test:coverage": "PERCY_LOGLEVEL=silent npx jest --coverage",
-    "test:ui": "npx jest --coverage --runInBand --detectOpenHandles --logHeapUsage --forceExit -- func -- e2e",
+    "test:ui": "npx jest --coverage --runInBand --detectOpenHandles --forceExit -- func -- e2e",
```
Comment (tmcconechy) [inline package.json:30]: Why remove this it was handy?

File: package.json [hunk 4]
```diff
@@ -46,59 +46,60 @@
...
+    "ts-loader": "^9.2.8",
+    "ts-node": "^10.7.0",
+    "typescript": "^4.6.3",
+    "webpack": "5.72.0",
     "webpack-bundle-analyzer": "^4.5.0",
     "webpack-cli": "^4.9.2",
-    "webpack-dev-server": "^4.9.2",
+    "webpack-dev-server": "^4.8.1"
+  },
+  "dependencies": {
```
Comment (tmcconechy) [inline package.json:102]: Should be a dev depenency

File: package.json [hunk 5]
```diff
@@ -46,59 +46,60 @@
...
+    "express": "^4.17.3",
+    "express-fileupload": "^1.3.1",
     "file-loader": "^6.2.0",
     "html-loader": "^3.1.0",
     "html-webpack-plugin": "^5.5.0",
     "htmlhint": "^1.1.4",
-    "ids-identity": "4.11.9",
-    "jest": "^28.1.1",
-    "jest-environment-jsdom": "^28.1.1",
+    "ids-identity": "4.11.6",
```
Comment (tmcconechy) [inline package.json:73]: Need to keep this one

**Review by tmcconechy — CHANGES_REQUESTED** [review]
A lot of extra rollbacks here. If you want to just fix it so it doesnt have those errors im fine with that. So then i would only expect a few dependencies to be altered.

A few other things in there

File: package.json (hunk 5 above)
Comment (danielortiz1982) [reply package.json:73]: @tmcconechy Thanks for the feedback, can please review what needs to removed and updated to make the `npm i` run with no errors or vulnerabilities. Also running `npm test` does not seem to be working properly... I am not sure if it is dependency related so let's open up another issue for this.

**Review by EdwardCoyle — COMMENTED** [review]
Running the build seemed to work but I couldn't use `npm run test`.  Seems like the Jest dependencies are missing:

Screen Shot 2022-06-09 at 9 35 27 AM

File: package.json (hunk 3 above)
Comment (jdilag90) [reply package.json:30]: --logHeapUsage was useful detecting memory leaks/limits. From our tests it didn't really affect testing speed. 

https://jestjs.io/docs/cli#--logheapusage

**Comment by tmcconechy** [comment]
As discussed will fix this a bit differently -  will make a new PR

**Pull request closed without merging**